### ISSUE TITLE
Binary table with columns containing strings not read properly in Astropy v4.2

### ISSUE THREAD

**Issue by Skyhawk172** [astropy/astropy#11275]
### Description

It appears like Astropy v4.2 is not reading binary table with columns containing strings properly, as I'm getting ones or zeros in those columns.  When downgrading to Astropy v4.0, the columns are read properly.

### Expected behavior
I expected the columns names "target_designation" and "gs_designation" to contain strings, such as:

```
target_designation     ra            decl     gs_designation 
-------------------------------------------------------------------------
 18010048+6835075    270.252039   68.585434   18003716+6833239 
 18010048+6835075    270.252039.  68.585434   18005781+6837508 
 18010048+6835075    270.252039.  68.585434   18005353+6837277
```

### Actual behavior
However, when running Astropy v4.2, the two columns of interest contains only ones:
```
target_designation   ra [1]    decl [1] gs_designation
------------------ ---------- --------- --------------
                 1 270.252039 68.585434              1
                 1 270.252039 68.585434              1
                 1 270.252039 68.585434              1
```

### Steps to Reproduce

1. Unzip the following file: [TEST_OTE07_cvz_north_NRCA3_FULL_5p0_150_dev_lajoiecp_13.fit.zip](https://github.com/astropy/astropy/files/5876251/TEST_OTE07_cvz_north_NRCA3_FULL_5p0_150_dev_lajoiecp_13.fit.zip)
2. Run the code lines below and compare results with the above expected/actual behaviors.
3. Downgrade to Astropy v4.0.0 using `conda install astropy=4.0.0` and run the same lines of code again.

```python
from astropy.table import Table

t = Table.read("./TEST_OTE07_cvz_north_NRCA3_FULL_5p0_150_dev_lajoiecp_13.fit")
print( t[:3]["target_designation", "ra", "decl","gs_designation"] )
```

### System Details

> import platform; print(platform.platform())
Darwin-19.6.0-x86_64-i386-64bit

> import numpy; print("Numpy", numpy.__version__)
Numpy 1.19.2

 > import astropy; print("astropy", astropy.__version__)
astropy 4.2

> import scipy; print("Scipy", scipy.__version__)
Scipy 1.5.2

> import matplotlib; print("Matplotlib", matplotlib.__version__)
Matplotlib 3.3.2

**Comment by pllim** [comment]
I think this is more of an `io.fits` issue. With 4.3.dev:

```python
>>> from astropy.io import fits
>>> pf = fits.open('TEST_OTE07_cvz_north_NRCA3_FULL_5p0_150_dev_lajoiecp_13.fit')
>>> pf[1].data.field('target_designation')
chararray(['1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1',
           '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1',
           '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1',
           '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1',
           '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1',
           '1', '1', '1', '1', '1', '1', '1'], dtype='<U1')
>>> pf.info()
Filename: TEST_OTE07_cvz_north_NRCA3_FULL_5p0_150_dev_lajoiecp_13.fit
No.    Name      Ver    Type      Cards   Dimensions   Format
  0  PRIMARY       1 PrimaryHDU       4   ()      
  1                1 BinTableHDU     46   67R x 12C   ['17A', '1J', '1D', '1D', '1D', '1D', '17A', '1J', '1D', '1D', '1D', '1D']
>>> pf[1].header
XTENSION= 'BINTABLE'           / C# FITS: 1/26/2021 8:20:12 PM                  
BITPIX  =                    8                                                  
NAXIS   =                    2 / Dimensionality                                 
NAXIS1  =                  106 / row width in bytes                             
NAXIS2  =                   67 / number of rows                                 
PCOUNT  =                    0                                                  
GCOUNT  =                    1                                                  
TFIELDS =                   12                                                  
TFORM1  = '17A     '                                                            
TDIM1   = '(1)     '                                                            
TFORM2  = '1J      '                                                            
TDIM2   = '(1)     '                                                            
TFORM3  = '1D      '                                                            
TDIM3   = '(1)     '                                                            
TFORM4  = '1D      '                                                            
TDIM4   = '(1)     '                                                            
TFORM5  = '1D      '                                                            
TDIM5   = '(1)     '                                                            
TFORM6  = '1D      '                                                            
TDIM6   = '(1)     '                                                            
TFORM7  = '17A     '                                                            
TDIM7   = '(1)     '                                                            
TFORM8  = '1J      '                                                            
TDIM8   = '(1)     '                                                            
TFORM9  = '1D      '                                                            
TDIM9   = '(1)     '                                                            
TFORM10 = '1D      '                                                            
TDIM10  = '(1)     '                                                            
TFORM11 = '1D      '                                                            
TDIM11  = '(1)     '                                                            
TFORM12 = '1D      '                                                            
TDIM12  = '(1)     '                                                            
TTYPE1  = 'target_designation'                                                  
TTYPE2  = 'target_objid'                                                        
TNULL2  =                  -99                                                  
TTYPE3  = 'ra      '                                                            
TTYPE4  = 'decl    '                                                            
TTYPE5  = 'target_k_m'                                                          
TTYPE6  = 'target_j_m'                                                          
TTYPE7  = 'gs_designation'                                                      
TTYPE8  = 'gs_objid'                                                            
TNULL8  =                  -99                                                  
TTYPE9  = 'gs_ra   '                                                            
TTYPE10 = 'gs_decl '                                                            
TTYPE11 = 'guide_star_j_m'                                                      
TTYPE12 = 'gs_distance'   
```

How was this file created?

**Comment by pllim** [comment]
If I change your `TDIMx` from `(1)` to the actual elements you are expecting, then I can see your actual values. For example:

```python
>>> from astropy.io import fits
>>> with fits.open(filename, mode='update') as pf:
...     pf[1].header['TDIM1'] = '(17)'
>>> pf = fits.open(filename)
>>> pf[1].data.field('target_designation')
chararray(['18010048+6835075', '18010048+6835075', '18010048+6835075',
           '18010048+6835075', '18010048+6835075', '18010048+6835075',
           '18010048+6835075', '18010048+6835075', '18010048+6835075',
           '18100283+6849131', '17280677+6414096', '17313976+6430391',
           '17313976+6430391', '17313976+6430391', '17305711+6506372',
           '17305711+6506372', '17305711+6506372', '17305711+6506372',
           '17305711+6506372', '17305711+6506372', '17441965+6537227',
           '17441965+6537227', '17441965+6537227', '17441965+6537227',
           '17344183+6655222', '17344183+6655222', '17344183+6655222',
           '17344183+6655222', '17352729+7005145', '17352729+7005145',
           '17352729+7005145', '17352729+7005145', '17352729+7005145',
           '17352729+7005145', '17352729+7005145', '18372852+6642338',
           '18372852+6642338', '18372852+6642338', '17542048+6404591',
           '17542048+6404591', '17542048+6404591', '18122139+6536173',
           '18122139+6536173', '18122139+6536173', '18122139+6536173',
           '18141226+6445083', '18141226+6445083', '18141226+6445083',
           '18141226+6445083', '18141226+6445083', '18141226+6445083',
           '18162135+6529381', '18162135+6529381', '18162135+6529381',
           '18201962+6412108', '18201962+6412108', '18201962+6412108',
           '17362594+6820220', '17362594+6820220', '17362594+6820220',
           '17362594+6820220', '17362594+6820220', '17362594+6820220',
           '17362594+6820220', '17362594+6820220', '17370028+6803336',
           '17370028+6803336'], dtype='<U17')
```

Not sure why it worked for older `astropy` version though. Perhaps other FITS experts could chime in.

**Comment by saimn** [comment]
The change is related to #10768 which fixed the use of `TDIM=(1)` (in v4.0.2), and as pointed by @pllim it's because of the use of `TDIM=(1)` with a char column. For the other columns it gives you a multidimensional column, where each element is an array of one element, but for chars it means that only the first character is represented.
